homeassistant/components/google/__init__.py

```diff
@@ -97,8 +101,8 @@
     {
         DOMAIN: vol.Schema(
             {
-                vol.Required(CONF_CLIENT_ID): cv.string,
-                vol.Required(CONF_CLIENT_SECRET): cv.string,
+                vol.Optional(CONF_CLIENT_ID): cv.string,
```

Comment:
These are still required if the user has a google section in the YAML CONFIG. It's the section that's optional. It already is that.

Reply:
I am not yet deprecating other config options in the schema as part of the move to application credentials. Is that an issue? Those need more config or options flow support or their own deprecation plan.

Reply:
Ok.

Reply:
Note that we haven't bumped the frontend version yet in core so this should probably wait for that.

Reply:
OK, good point.  Let's do this: I'll revert this part, leave it as required, and revisit making it optional as a separate PR. There is not a specific test of that case anyway, and it is unclear how much it really matters.

Reply:
Sounds good!